### PR TITLE
Add learning rate in XGBoost example and update readme for same

### PR DESCRIPTION
**Pull request by shubham769**
## What changes are proposed in this pull request?

I have added the learning rate in xgboost example as the parameter play a huge role in optimization, moreover updated the readme by adding a way to do experiments and track them on  mlflow UI
## How is this patch tested?
I have tested this on my local machine and it is running fine
(Details)

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

This PR will make changes in xgboost example, which consists of the addition of an important parameter `learning rate ` which plays a huge role in optimization. Moreover, Readme for the example is changed for the same modification and adding a way to do experiments and performance tracking with mlflow.

### What component(s) does this PR affect?

- [ ] UI
- [ ] Command Line Interface
- [ ] API
- [x] Examples
- [x] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Model Registry
- [ ] Scoring
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by shubham769** [comment]
cc @smurching  @arocketman @tomasatdatabricks

**Comment by shubham769** [comment]
@tomasatdatabricks

**Comment by shubham769** [comment]
@pogil , can you please review this?

**Comment by shubham769** [comment]
@pogil, I have added the released note above in or comments, please can you review it?

**Comment by shubham769** [comment]
@sueann , can you please look into this?

File: examples/xgboost/README.md [hunk 1]
```diff
@@ -5,8 +5,18 @@ This example trains an XGBoost classifier with the iris dataset and logs hyperpa
 ## Running the code
 
 ```
-python train.py --colsample-bytree 0.8 --subsample 0.9
+python train.py --eta 0.2 --colsample-bytree 0.8 --subsample 0.9
 ```
+After that try experiments with different parameters like:
```
Comment (sueann) [inline examples/xgboost/README.md:10]:
```suggestion
You can try experimenting with different parameter values like:
```

File: examples/xgboost/README.md [hunk 2]
```diff
@@ -5,8 +5,18 @@ This example trains an XGBoost classifier with the iris dataset and logs hyperpa
...
+```
+python train.py --eta 0.4 --colsample-bytree 0.7 --subsample 0.8
+```
+
+Then you can open the mlflow ui to compare your logs and track the experiments by:
```
Comment (sueann) [inline examples/xgboost/README.md:15]:
```suggestion
Then you can open the MLflow UI to track the experiments and compare your runs via:
```

File: examples/xgboost/README.md [hunk 3]
```diff
@@ -5,8 +5,18 @@ This example trains an XGBoost classifier with the iris dataset and logs hyperpa
...
+```
+mflow ui
```
Comment (sueann) [inline examples/xgboost/README.md:17]:
```suggestion
mlflow ui
```

File: examples/xgboost/train.py [hunk 4]
```diff
@@ -43,6 +45,7 @@ def main():
         params = {
             'objective': 'multi:softprob',
             'num_class': 3,
+            'learning_rate': args.eta,
```
Comment (sueann) [inline examples/xgboost/train.py:48]: curious why we made the parser argument "eta" and use "learning_rate" to specify the parameter to xgboost. they are interchangeably used by xgboost, right? is "eta" more commonly used by xgboost users?

**Review by sueann — COMMENTED** [review]
Thank you for your contribution! This looks mostly good to me. 

* I have added some inline suggestions for small wording changes, and had one question about the parameter name.
* Could we add the new parameter to the example tests at https://github.com/mlflow/mlflow/blob/master/tests/examples/test_examples.py#L29 and https://github.com/mlflow/mlflow/blob/master/tests/examples/test_examples.py#L44?
  * I don't think the tests will be run automatically by travis for the PR, so you'll need to manually test: e.g. https://github.com/mlflow/mlflow/blame/fcef783d4019230b140d3c97b92bc3db33a031f1/travis/stage-python3.sh#L17

Thanks again!

File: examples/xgboost/train.py (hunk 4 above)
Comment (shubham769) [reply examples/xgboost/train.py]: @sueann, I have changed the argument from "eta" to "learning_rate" and No, "learning_rate" is commonly used by users. I think now it's looking good after the change.

**Comment by shubham769** [comment]
@sueann , I have added the parameter in tests and also changed the parameter 'eta to 'learning_rate'.

File: examples/xgboost/train.py [hunk 5]
```diff
@@ -15,6 +15,8 @@
 
 def parse_args():
     parser = argparse.ArgumentParser(description='XGBoost example')
+    parser.add_argument('--learning_rate', type=float, default=0.3,
+                        help='learning rate to update step size at each boosting step (default=0.3)')
```
Comment (harupy) [inline examples/xgboost/train.py:19]:
```suggestion
                        help='learning rate to update step size at each boosting step (default: 0.3)')
```

Comment (harupy) [reply examples/xgboost/train.py:19]: `colsample-bytree` and `subsample` use this format.

**Comment by harupy** [comment]
I think [examples/xgboost/MLproject](https://github.com/mlflow/mlflow/blob/master/examples/xgboost/MLproject) should be updated too.

**Comment by harupy** [comment]
@shubham769 Can you fix the lint error?

**Comment by harupy** [comment]
@shubham769 Can you fix this lint error?

https://github.com/mlflow/mlflow/pull/2766/checks?check_run_id=681044249#step:5:8

**Comment by shubham769** [comment]
> @shubham769 Can you fix this lint error?
> 
> https://github.com/mlflow/mlflow/pull/2766/checks?check_run_id=681044249#step:5:8
@harupy I have fix lint error, please review

**Comment by harupy** [comment]
LGTM

File: examples/xgboost/train.py [hunk 6]
```diff
@@ -15,6 +15,8 @@
 
 def parse_args():
     parser = argparse.ArgumentParser(description='XGBoost example')
+    parser.add_argument('--learning_rate', type=float, default=0.3,
```
Comment (sueann) [inline examples/xgboost/train.py:18]: could you change this to `--learning-rate` to match the convention here? (It shouldn't change line 48's `args.learning_rate,` though.) thanks!

Comment (shubham769) [reply examples/xgboost/train.py]: done @sueann

File: examples/xgboost/MLproject [hunk 7]
```diff
@@ -3,9 +3,11 @@ conda_env: conda.yaml
 entry_points:
   main:
     parameters:
+      learning_rate: {type: float, default: 0.3}
       colsample-bytree: {type: float, default: 1.0}
       subsample: {type: float, default: 1.0}
     command: |
         python train.py \
+          --learning_rate={learning_rate} \
```
Comment (sueann) [inline examples/xgboost/MLproject:11]:
```suggestion
          --learning-rate={learning_rate} \
```

File: examples/xgboost/MLproject [hunk 8]
```diff
@@ -3,9 +3,11 @@ conda_env: conda.yaml
 entry_points:
   main:
     parameters:
+      learning_rate: {type: float, default: 0.3}
```
Comment (sueann) [inline examples/xgboost/MLproject:6]:
```suggestion
      learning-rate: {type: float, default: 0.3}
```

File: examples/xgboost/MLproject [hunk 9]
```diff
@@ -3,9 +3,11 @@ conda_env: conda.yaml
...
       colsample-bytree: {type: float, default: 1.0}
       subsample: {type: float, default: 1.0}
     command: |
         python train.py \
+          --learning-rate={learning_rate} \
```
Comment (sueann) [inline examples/xgboost/MLproject:11]:
```suggestion
          --learning-rate={learning-rate} \
```

**Review by sueann — COMMENTED** [review]
sorry about the million tiny edits! i think the way we test it, it's better the project param matches what's passed into the `train.py` script. i think when we run the tests with the changes it's likely not passing in the parameters correctly, but doing so silently.

**Comment by shubham769** [comment]
> sorry about the million tiny edits! i think the way we test it, it's better the project param matches what's passed into the `train.py` script. i think when we run the tests with the changes it's likely not passing in the parameters correctly, but doing so silently.

@sueann, no problem at all, although I had tested it manually, but yes, I am agree with you.
I have made the changes according to the suggestions, can you please review?

**Review by sueann — APPROVED** [review]
Awesome - thanks! Will merge shortly.
